src/main/java/fr/minuskube/inv/InventoryManager.java

```diff
@@ -123,17 +124,21 @@ public void onInventoryClick(InventoryClickEvent e) {
             if (!inventories.containsKey(p.getUniqueId()))
                 return;
 
-            if (e.getAction() == InventoryAction.COLLECT_TO_CURSOR || e.getAction() == InventoryAction.MOVE_TO_OTHER_INVENTORY) {
-                e.setCancelled(true);
-                return;
-            }
-
-            if (e.getAction() == InventoryAction.NOTHING && e.getClick() != ClickType.MIDDLE) {
-                e.setCancelled(true);
-                return;
+            // Restrict putting items from the bottom inventory into the top inventory
+            Inventory clickedInventory = e.getClickedInventory();
```

Comment:
Maybe need a null-check to avoid inventory-outside clicking.

Reply:
Referential equality is in and of itself a null check. Not necessary.

Reply:
Yea I know, but maybe in the future someone use the clickedInventory without null-check. I don't know maybe I'm a little bit paranoid :D

Reply:
I think that's a bridge to cross when somebody gets to it :)